java/dev/checkstyle/checkstyle.xml

```diff
@@ -205,7 +205,7 @@
         <module name="CustomImportOrder">
             <property name="specialImportsRegExp" value="com.google"/>
```

Comment:
I'm not really sure why we have `com.google` as a special import, anyone know?

Reply:
This is adopted from google style guide and that's why it is a special import.

Reply:
This seems to be updated:
https://github.com/checkstyle/checkstyle/commit/81ad4595fe3a916551d73e467044c87d8462dba6#diff-c55cd603ef6597463971db3f33f4f4c8

I think we can update the CustomImportOrder section to be the same as in:

https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L193

Reply:
> I think we can update the CustomImportOrder section to be the same as in

That just has one single group, so basically there is no grouping rules unless I misunderstand.  Is that what your preference is?

I like having some order to the groups and since we don't need "com.google" as a special import maybe we can use it for `io.netty.buffer` since we have some source files in that package.  So I would propose this order:

Static imports
Java standard imports
Third-party imports
Special imports using io.netty.buffer (for ArrowBuf, etc)
Same package 4 places (org.apache.arrow.vector, etc) although 3 place might be better..

I'm not too particular about the order, but I think *some* order is good and definitely keep arrow imports together and not mixed in with other 3rd party imports. wdyt @icexelloss ?

Reply:
Your proposed order looks fine to me.